src/core/kmod-setup.c

```diff
@@ -166,13 +165,12 @@ int kmod_setup(void) {
                 { "tpm",                        "/sys/class/tpmrm",          false, false, efi_has_tpm2       },
 #endif
         };
-        _cleanup_(kmod_unrefp) struct kmod_ctx *ctx = NULL;
-        unsigned i;
 
         if (have_effective_cap(CAP_SYS_MODULE) <= 0)
                 return 0;
 
-        for (i = 0; i < ELEMENTSOF(kmod_table); i++) {
+        _cleanup_(kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        for (unsigned i = 0; i < ELEMENTSOF(kmod_table); i++) {
```

Comment:
This can be switched to FOREACH_ARRAY ;-)

Reply:
also, size_t is the right type when iterating through memory, it's also the type ELEMENTSOF() returns, hence it's weird dropping half of the bits on the ground for no real reason